src/extension.ts

```diff
@@ -93,10 +95,68 @@ export function activate(context: ExtensionContext) {
     const languageClient = createLangServer(context);
     let hubWebviewPanel: WebviewPanel | undefined = undefined;
 
+    let tree = new DiagnosticTreeView();
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Split out scan functionality into a separate file

This file is starting to feel unwieldy, it'd be good to keep all scanning together and apart from everything else

Reply:
have improved the file a fair bit - not sure how to do the scan split